test/integration/driver_install_or_update_darwin_test.go

```diff
@@ -0,0 +1,124 @@
+			expectedVersion: version.GetVersion(),
+		},
+	}
+
+	sudoPath, err := exec.LookPath("sudo")
+	if err != nil {
+		t.Fatalf("No sudo in path: %v", err)
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			mkDir, drvPath, err := tempMinikubeDir(tc.name, tc.path)
+			if err != nil {
+				t.Fatalf("Failed to prepare tempdir. test: %s, got: %v", tc.name, err)
+			}
+			defer func() {
+				if err := os.RemoveAll(mkDir); err != nil {
+					t.Errorf("Failed to remove mkDir %q: %v", mkDir, err)
+				}
+			}()
+
+			// start "minikube start --download-only --interactive=false --driver=hyperkit --preload=false"
+			cmd := exec.Command(Target(), "start", "--download-only", "--interactive=false", "--driver=hyperkit", "--preload=false")
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stdout
+			// set PATH=<tmp_minikube>/bin:<path_to_sudo>
+			//     MINIKUBE_PATH=<tmp_minikube>
+			cmd.Env = append(os.Environ(),
+				fmt.Sprintf("PATH=%v%c%v", filepath.Dir(drvPath), filepath.ListSeparator, filepath.Dir(sudoPath)),
+				"MINIKUBE_HOME="+mkDir)
+			if err = cmd.Run(); err != nil {
+				t.Fatalf("failed to run minikube. got: %v", err)
+			}
+
+			upgradedVersion, err := driverVersion(drvPath)
+			if err != nil {
+				t.Fatalf("failed to check driver version. got: %v", err)
+			}
+
+			if upgradedVersion != tc.expectedVersion {
+				t.Fatalf("invalid driver version. expected: %v, got: %v", tc.expectedVersion, upgradedVersion)
+			}
+		})
+	}
+}
+
+func driverVersion(path string) (string, error) {
+	output, err := exec.Command(path, "version").Output()
+	if err != nil {
+		return "", err
+	}
+
+	var resultVersion string
+	_, err = fmt.Sscanf(string(output), "version: %s\n", &resultVersion)
+	if err != nil {
+		return "", err
+	}
+	return resultVersion, nil
+}
+
+func tempMinikubeDir(name, driver string) (string, string, error) {
+	temp, err := ioutil.TempDir("", name)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to create tempdir: %v", err)
+	}
+
+	mkDir := filepath.Join(temp, ".minikube")
+	mkBinDir := filepath.Join(mkDir, "bin")
+	err = os.MkdirAll(mkBinDir, 0777)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to prepare tempdir: %v", err)
+	}
+
+	pwd, err := os.Getwd()
+	if err != nil {
+		return "", "", fmt.Errorf("failed to get working directory: %v", err)
+	}
+
+	testDataDriverPath := filepath.Join(pwd, driver, "docker-machine-driver-hyperkit")
+	if _, err = os.Stat(testDataDriverPath); err != nil {
+		return "", "", fmt.Errorf("expected driver to exist: %v", err)
+	}
+
+	// copy driver to temp bin
+	testDriverPath := filepath.Join(mkBinDir, "docker-machine-driver-hyperkit")
+	if err = ioext.CopyFile(testDataDriverPath, testDriverPath, false); err != nil {
+		return "", "", fmt.Errorf("failed to setup current hyperkit driver: %v", err)
+	}
+	// change permission to allow driver to be executable
+	if err = os.Chmod(testDriverPath, 0777); err != nil {
```

Comment:
is this same chmod number we have on our driver binary ? if not reduce from  755 to what we have in reality.

Reply:
Yep, good point! Done.

Reply:
dont see the updated code here

Reply:
oops, my bad, changed the wrong line. Fixed